### ISSUE TITLE
Будут ли сюрпризы?

### ISSUE THREAD

**Issue by ewgRa** [MailRuChamps/hlcupdocs#147]
Для group сказано: "Перед тем как выполнять группировку необходимо выполнить выборку как в предыдущем запросе, но по конкретным значениям, а не по предикатам"

В текущих данных нету ни одного запроса к group, в котором например есть sname фильтр.

Вопрос, это сохранится? Или возможны сюрпризы?

Почему это все дело интересует - теоретически такой запрос возможен судя по заданию, но если допустить что на практике он не придет - это можно использовать.

И тут получается более законопослушные участники, которые закрывают и такие запросы будут чувствовать несправедливость.

**Comment by ewgRa** [comment]
В общем да, вопросов все больше.

Чем больше оптимизаций, тем более хрупким становится решение.
Например число уникальных "interests" сейчас вмещается в один байт, если задаться целью и добавить в систему много новых интересов - то скорее всего решения начнут ломаться.

**Comment by F10PPY** [comment]
Да, хотелось бы знать, хотя бы, будут ли данные для финала создаваться изменённым генератором или нет? То есть например, если GET и POST сейчас можно проверять по одной букве, не будут ли эти параметры искажены рандомно в финальных данных?

**Comment by SannikovDmitry** [comment]
Изменений в финале в правилах не будет.

**Comment by ewgRa** [comment]
@SannikovDmitry в правилах понятно, а в системе проверки?

выше как минимум два конкретных примера, на самом деле их больше гораздо.
